fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/metadata/KeyExpressionTest.java

```diff
@@ -725,103 +730,132 @@ public void testSplitConcat() throws Exception {
-                        false},
-                {concat(field("a"), VERSION),
+                        false),
+                Arguments.of(concat(field("a"), VERSION),
                         concat(field("a"), field("b")),
-                        false},
-                {concat(field("a"), VERSION),
+                        false),
+                Arguments.of(concat(field("a"), VERSION),
                         concat(field("a"), VERSION),
-                        true},
-                {field("a").split(3),
+                        true),
+                Arguments.of(field("a").split(3),
                         concat(field("a").split(3), field("b"), field("c")),
-                        true},
-                {field("a").split(2),
+                        true),
+                Arguments.of(field("a").split(2),
                         field("a").split(3),
-                        false},
-                {field("a").split(3),
+                        false),
+                Arguments.of(field("a").split(3),
                         field("a").split(2),
-                        false}
-        };
-
-        for (Object[] test : tests) {
-            assertEquals(test[2],
-                    ((KeyExpression) test[0]).isPrefixKey((KeyExpression) test[1]),
-                    "\nPrefix: " + test[0] + "\nKey: " + test[1]);
-        }
+                        false),
+                Arguments.of(keyWithValue(concat(field("a"), field("b")), 1),
+                        concat(field("a"), field("c")),
+                        true),
+                Arguments.of(keyWithValue(concat(field("a"), field("b")), 1),
+                        field("a"),
+                        true),
+                Arguments.of(keyWithValue(concat(field("a"), field("b"), field("c")), 2),
+                        concat(field("a"), field("c")), false),
+                Arguments.of(concat(field("a"), field("b")),
+                        keyWithValue(concat(field("a"), field("b"), field("c")), 2),
+                        true),
+                Arguments.of(concat(field("a"), field("b")),
+                        keyWithValue(concat(field("a"), field("b")), 1),
+                        false),
+                Arguments.of(field("a", FanType.FanOut).nest(field("b")),
+                        nestedKeyWithValue,
+                        true),
+                Arguments.of(field("a", FanType.FanOut).nest(concat(field("b"), field("c"))),
+                        nestedKeyWithValue,
+                        true),
+                Arguments.of(field("a", FanType.FanOut).nest(
+                        concat(field("b"), field("c"), field("d"))),
+                        nestedKeyWithValue,
+                        false),
+                Arguments.of(concat(field("a", FanType.FanOut).nest(
+                        field("b")), field("a", FanType.FanOut).nest("b")),
```

Comment:
Or it could be another test.
```suggestion
                        field("b")), field("a", FanType.FanOut).nest("c")),
```

Reply:
Added.